source/GameInterface/Services/Villages/Patches/ChangeVillageStatePatch.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Services.Villages.Messages;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Villages.Patches
+{
+    [HarmonyPatch(typeof(ChangeVillageStateAction), "ApplyInternal")]
+    public class ChangeVillageStatePatch
+    {
+        private static AllowedInstance<Village> _allowedInstance;
+        private static readonly MethodInfo _applyInternal = typeof(ChangeVillageStateAction).GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
I don't remember if this works with statics but give it a try. If not let me know and keep as is.

I want any calls that happen a lot to use delegates as reflection invoke is fairly slow

```suggestion
        private static MethodInfo _applyInternal => typeof(ChangeVillageStateAction).GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static);
        private static readonly Action<Village, Village.VillageStates, MobileParty> ApplyInternalDelegate = _applyInternal .BuildDelegate<Action<Village, Village.VillageStates, MobileParty>>();
```